### DISCUSSION TITLE
Feature Request: `ChaosInfra` resource to manage the lifecycle of Chaos Infrastructures

### DISCUSSION THREAD

**Discussion by smitthakkar96** [litmuschaos/chaos-operator#487] [Ideas]
## Current Challenge

The current onboarding and offboarding process for Chaos Infra involves a manual sequence of steps, including the `registerInfra` call for obtaining essential information like `token`, `infraID`, and `manifest`. To offboard, the process requires manual resource cleanup via the `deleteInfra` call. This manual intervention becomes cumbersome and impractical when dealing with a large-scale deployment of Litmus execution plane across numerous clusters and namespaces.

## Proposed Solution

To streamline the management of Chaos Infra at scale, we suggest introducing a custom resource specifically designed for Chaos Infra lifecycle management. Similar to the ChaosEngine resource, which abstracts the complexities of Argo Workflows, this new resource would encapsulate the entire lifecycle of Chaos Infra and its associated Kubernetes resources.

**Comment by smitthakkar96** [comment]
If this feature request makes sense, I am happy to contribute this as it would simplify the adoption of Litmus in our org. Also, I am wondering if it would make sense to have a resource for project and environment entities

**Comment by neelanjan00** [comment]
This sounds good! Let's hear more opinions on this from others, I am moving it to Discussion for now. If you'd like to contribute, then it will be good to start with a design spec.

**Comment by smitthakkar96** [comment]
@neelanjan00 we developed the first version internally as part of efforts to reduce toil in onboarding new users to Chaos Platform (Litmus), it currently offers `Project`, `Environment` and `LitmusChaosExecutionPlane` resources. If there is an interest, we can do a demo in the next contributors call and discuss the possibility of potentially open-sourcing it. Let me know what you think.

**Reply by neelanjan00** [reply]
That sounds like a good idea. CC: @Saranya-jena

**Comment by neelanjan00** [comment]
@ispeakc0de @ksatchit would like to hear your opinion about this, WDYT?
